### ISSUE TITLE
Flutter plugin crashes after using FlutterInspector

### ISSUE THREAD

**Issue by ohir** [flutter/flutter#15786]
## Steps to Reproduce
1. Create fresh 'hello' project from cli:
```
% flutter create mojeagd
% cd mojeagd
% flutter run # to get apk built
q
```
2. `Android Studio -> Open Project -> ...mojeagd -> Run(button)`

App is running in emulator, I fiddled with Flutter Inspector a bit switching
between manuals and AS/emulator. First the app in emulator 'froze', then

3. I pushed `Android Studio -> Stop(button)`, then flutter plugin crashed with
`"Flutter threw an uncaught Empty Throwable"` message.

## Logs

Run your application with `flutter run` and attach all the log output.
```
Launching lib/main.dart on Android SDK built for x86 in debug mode...
Initializing gradle...
Resolving dependencies...
Running 'gradlew assembleDebug'...
Built build/app/outputs/apk/debug/app-debug.apk (25.0MB).
I/FlutterActivityDelegate(26689): onResume setting current activity to this
Syncing files to device Android SDK built for x86...
```
[...cut hot reload message... App quit with q on console]
`Application finished.`

Run `flutter analyze` and attach any output of that command also.
```
Analyzing /testbed/flutter/mojeagd...
No issues found!
```
## Flutter Doctor

Paste the output of running `flutter doctor -v` here.
```
[✓] Flutter (Channel beta, v0.1.5, on Linux, locale pl_PL.utf8)
    • Flutter version 0.1.5 at /testbed/lib/flutter
    • Framework revision 3ea4d06340 (4 weeks ago), 2018-02-22 11:12:39 -0800
    • Engine revision ead227f118
    • Dart version 2.0.0-dev.28.0.flutter-0b4f01f759

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.2)
    • Android SDK at /testbed/lib/Android/Sdk
    • Android NDK at /testbed/lib/Android/Sdk/ndk-bundle
    • Platform android-27, build-tools 26.0.2
    • ANDROID_HOME = /testbed/lib/Android/Sdk
    • Java binary at: /testbed/lib/Android/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[✓] Android Studio (version 3.0)
    • Android Studio at /testbed/lib/Android/android-studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[✓] Connected devices (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 6.0 (API 23) (emulator)
```

## idea.log for the crash
```
2018-03-21 08:30:15,775 [of flutter]   WARN - o.flutter.run.daemon.DaemonApi - Flutter process returned an error: {"id":1,"error":"TimeoutException: Request to Dart VM Service timed out: ext.flutter.debugPaint({enabled: true, isolateId: isolates/1071090508})"} 
2018-03-21 08:30:15,775 [of flutter]   WARN - o.flutter.run.daemon.DaemonApi - received a response for a request that wasn't sent: 1 
2018-03-21 08:30:15,775 [of flutter]   INFO - r.run.daemon.DaemonConsoleView - [{"id":1,"error":"TimeoutException: Request to Dart VM Service timed out: ext.flutter.debugPaint({enabled: true, isolateId: isolates/1071090508})"}] 
2018-03-21 08:39:33,633 [ Thread-15]  ERROR - er.inspector.EvalOnDartLibrary - Error evaluating expression:
Unhandled exception:
'dart:collection-patch/dart:collection/compact_hash.dart': Failed assertion: line 529: '(hashPattern & _usedData) == 0': is not true.
#0      _AssertionError._doThrowNew (dart:core-patch/dart:core/errors_patch.dart:37)
#1      _AssertionError._throwNew (dart:core-patch/dart:core/errors_patch.dart:33)
#2      _CompactLinkedHashSet.add (dart:collection-patch/dart:collection/compact_hash.dart:529)
#3      WidgetInspectorService.toId (package:flutter/src/widgets/widget_inspector.dart:198)
#4      WidgetInspectorService._nodeToJson (package:flutter/src/widgets/widget_inspector.dart:357)
#5      WidgetInspectorService._nodesToJson.<anonymous closure> (package:flutter/src/widgets/widget_inspector.dart:373)
#6      MappedListIterable.elementAt (dart:_internal/iterable.dart:414)
#7      ListIterable.toList (dart:_internal/iterable.dart:219)
#8      WidgetInspectorService._nodesToJson (package:flutter/src/widgets/widget_inspector.dart:373)
#9      WidgetInspectorService.getProperties (package:flutter/src/widgets/widget_inspector.dart:380)
#10     Eval (evaluate:source:1:39)
#11     Eval (evaluate:source:1:4)
 
com.intellij.openapi.diagnostic.Logger$EmptyThrowable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:134)
	at io.flutter.inspector.EvalOnDartLibrary$2.received(EvalOnDartLibrary.java:121)
	at org.dartlang.vm.service.VmService.forwardResponse(VmService.java:410)
	at org.dartlang.vm.service.VmServiceBase.processResponse(VmServiceBase.java:432)
	at org.dartlang.vm.service.VmServiceBase$1.onMessage(VmServiceBase.java:99)
	at de.roderick.weberknecht.WebSocketReceiver.run(WebSocketReceiver.java:65)
2018-03-21 08:39:33,635 [ Thread-15]  ERROR - er.inspector.EvalOnDartLibrary - Android Studio 3.0.1  Build #AI-171.4443003 
2018-03-21 08:39:33,636 [ Thread-15]  ERROR - er.inspector.EvalOnDartLibrary - JDK: 1.8.0_152-release 
2018-03-21 08:39:33,636 [ Thread-15]  ERROR - er.inspector.EvalOnDartLibrary - VM: OpenJDK 64-Bit Server VM 
2018-03-21 08:39:33,636 [ Thread-15]  ERROR - er.inspector.EvalOnDartLibrary - Vendor: JetBrains s.r.o 
2018-03-21 08:39:33,636 [ Thread-15]  ERROR - er.inspector.EvalOnDartLibrary - OS: Linux 
2018-03-21 08:39:33,636 [ Thread-15]  ERROR - er.inspector.EvalOnDartLibrary - Last Action: Stop 
2018-03-21 08:39:33,770 [of flutter]   INFO - r.run.daemon.DaemonConsoleView - [{"id":2,"result":true}] 
2018-03-21 08:39:33,770 [of flutter]   INFO - r.run.daemon.DaemonConsoleView - [{"event":"app.stop","params":{"appId":"358f702b-44f2-4919-80e0-2aea04f2170e"}}] 
2018-03-21 08:39:33,804 [r: flutter]   INFO - .flutter.run.daemon.FlutterApp - StopApp mojeagd (Run) 
2018-03-21 08:39:33,840 [J pool 3/4]   WARN - ice.DartVmServiceDebugProcessZ - Failed to send request: {"id":"601","method":"evaluate","params":{"isolateId":"isolates/1071090508","targetId":"libraries/@323171358","expression":"WidgetInspectorService.instance.getSelectedRenderObject(\"inspector-1054\", \"intellij_inspector_1\")"}} 
de.roderick.weberknecht.WebSocketException: error while sending text data: not connected
	at de.roderick.weberknecht.WebSocket.send(WebSocket.java:182)
	at org.dartlang.vm.service.internal.WebSocketRequestSink.add(WebSocketRequestSink.java:43)
	at org.dartlang.vm.service.VmServiceBase.request(VmServiceBase.java:298)
	at sun.reflect.GeneratedMethodAccessor376.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.flutter.inspector.EvalOnDartLibrary.callVmServiceRequest(EvalOnDartLibrary.java:97)
	at io.flutter.inspector.EvalOnDartLibrary.evaluateHelper(EvalOnDartLibrary.java:190)
	at io.flutter.inspector.EvalOnDartLibrary.lambda$null$0(EvalOnDartLibrary.java:110)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:656)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:632)
	at java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:443)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```
last line (`2018-03-21 08:39:33,840 [J pool 3/4] ...` is repeated 7 times with different `J pool` and ` {"id":` numbers.

**Comment by chinmaygarde** [comment]
cc @devoncarew

**Comment by devoncarew** [comment]
@jacob314, I assume this will be fixed by https://github.com/flutter/flutter-intellij/pull/1980?

**Comment by jacob314** [comment]
This should be fixed by flutter/flutter-intellij#1980.
This is an identity hash map so the only way I can think of that it fails on that assert is if there is a bug in the identity hash map class (unlikely) or if the observatory protocol resulted in us making one insert into the hash map while another insert was in progress causing this strange behavior.
